### DISCUSSION TITLE
Rotation aix for cartesian kinematic

### DISCUSSION THREAD

**Discussion by andeksan** [Paciente8159/uCNC#623] [General]
I found this repository very modern and convenient compared with other similar repos. I already played with configuration and created simple 3-AIX CNC system with Mega 2560. Currently i am searching a way to add 4th rotation AIX-A. Is there way to make it?

**Comment by Paciente8159** [comment]
Hi there.
Sure. Change the AXIS_COUNT option to 4 on cnc_config.h
On the Mega2560 the extra steppers pinout is already configured by default.

**Reply by andeksan** [reply]
Hi,
Yes, did it before and it adds only linear aix. But in my case i need angular aix. 
I use UGS for communication. For example if i send a command **G1 A22.500 F141.4**. The motor makes more than 20 rotations. Obvious it is linear type of aix.
In my expectation it should make 1/4 rotation.
Important note: I use Shield v3 with mega 2560.

**Reply by Paciente8159** [reply]
Well you need to configure the step/mm setting $103 to match your needs.
In this case instead of steps/mm you need to think in steps/deg.
For example if a full rotation of your A Axis takes 3200 steps you need to set $103 to 3200/360 =~ 8.88889

`$103=8.88889`

**Reply by Paciente8159** [reply]
Note that rotational axis will also work like any normal axis depending on the distance mode (G90-absolute or G91-relative).
I've just added an option in PR #624 to allow rotational axis to always work on relative mode, independent of the distance mode enabled.

**Reply by andeksan** [reply]
thanks, i think i got idea.
